addons/sourcemod/scripting/surftimer/commands.sp

```diff
@@ -4153,11 +4153,15 @@ public Action Admin_FixBot(int client, int args)
 
 public Action Command_GiveKnife(int client, int args)
 {
-	if (IsPlayerAlive(client))
+	if (IsPlayerAlive(client) && (GetPlayerWeaponSlot(client, CS_SLOT_KNIFE) != -1))
 	{
-		GivePlayerItem(client, "weapon_knife");
+		CPrintToChat(client, "%t", "AlreadyHasKnife", g_szChatPrefix);
```

Comment:
eeeee i think here it should give the weapon

Reply:
It doesnt work that way because != -1

Entity index on success, -1 if no weapon existed.